lib/_stream_writable.js

```diff
@@ -191,14 +190,13 @@ function validChunk(stream, state, chunk, cb) {
   if (er) {
     stream.emit('error', er);
     process.nextTick(cb, er);
-    valid = false;
+    return false;
   }
-  return valid;
+  return true;
 }
```

Comment:
There might be a slight benefit in having a single return point. Not sure if it is visibile here or not.